### PR TITLE
Add MACSettings.DesiredMaxEIRP field

### LINKED ISSUE #4090: Allow configuring the MaxEIRP as part of device MAC settings
#### Summary

Allow configuring MaxEIRP for devices as part of the MAC settings

#### Why do we need this?

Currently, MaxEIRP can only be set in `MACState.DesiredParameters`. This is OK for ABP devices, but impossible for OTAA devices, since MAC state can only be configured if the device already has a session.

#### What is already there? What do you see now?

MaxEIRP is set in `MACState.DesiredParameters`. When initializing the MAC state of a device, the Network Server will use the default MaxEIRP setting from the respective band (https://github.com/TheThingsNetwork/lorawan-stack/blob/0e95225c3688a6d42e4f3bd2463f38258863bb91/pkg/networkserver/mac/utils.go#L221-L228).

This is also related to the Device Repository, where `maxEIRP` can be configured per device profile.

#### What is missing? What do you want to see?

Ability to configure MaxEIRP for OTAA devices.

#### Environment

3.12.1

#### How do you propose to implement this?

Introduce `MACSettings.MaxEIRP`. Network Server respects that value (if set) when initializing a new MAC state.

#### How do you propose to test this?

Unit tests, creating devices from the device repository

#### Can you do this yourself and submit a Pull Request?

Yes, but I am not sure if this is the correct way to do it. Adding to 3.12.2 because this needs to be addressed along with #4086

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #4090 

#### Changes

- Introduce `DeviceEIRPValue` struct (containing an enum value of type `DeviceEIRP`)
- Introduce `MACSettings.DesiredMaxEirp` field.
- Consider MaxEirp value for `mac.DeviceDesiredMaxEIRP`
- Miscellaneous changes required for validations and setting the DesiredMaxEirp field from the CLI.

#### Testing

- Unit tests
- Setting `desired_max_eirp` field from the CLI and verify that the `DesiredParameters.MaxEIRP` value is set properly after device joins.

##### Regressions

There should not be any, the default value for the new setting is `nil` which preserves the old behaviour.
 
#### Notes for Reviewers

- Not using gogoproto, so the field is `DesiredMaxEirp` instead of `DesiredMaxEIRP`.
- The validation and `FieldIsZero()` parts probably need a review from @rvolosatovs to make sure that I did not miss anything.

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [X] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by johanstokking — APPROVED** [review]
LGTM

File: pkg/ttnpb/testdata/ttnpb_encoding_golden.md
Comment (htdvisser) [inline pkg/ttnpb/testdata/ttnpb_encoding_golden.md:855]: I would also expect `ttnpb.DeviceEIRPValue` lines here

**Review by htdvisser — COMMENTED** [review]
I really don't know much about how these custom wrappers work, so maybe good to wait for review from @rvolosatovs.

**Review by adriansmares — APPROVED** [review]
Nothing to add.

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**
